nicegui/page.py

```diff
@@ -105,11 +105,24 @@ async def decorated(*dec_args, **dec_kwargs) -> Response:
             with Client(self) as client:
                 if any(p.name == 'client' for p in inspect.signature(func).parameters.values()):
                     dec_kwargs['client'] = client
+                if any(p.name == 'request_data' for p in inspect.signature(func).parameters.values()):
+                    url = request.url
+                    dec_kwargs['request_data'] = {'client':
```

Comment:
Why is the introduction of a `request_data` parameter necessary?

Reply:
It is required by async def root_page(request_data=None) in the class SinglePageRouterConfig to extract the query parameters and the original URL path.

Reply:
I see. In 2.0 we will provide a `client.request` property which would make this obsolete (see https://github.com/zauberzeug/nicegui/pull/2399). How can we avoid introducing this workaround as public API now? Maybe we could merge everything except the ip-property change to 1.4...